src/Core/Timer/BasicTimer.cs

```diff
@@ -1,44 +1,143 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
-namespace Chronos.Timer.Core
+namespace Chronos.Timer.Core.Timer
 {
-    //TODO Replace with real implementation
     public class BasicTimer : ITimer
     {
-        public TimeSpan ElapsedTime => throw new NotImplementedException();
+        #region private
+        
+        private SortedList<TimeSpan, ITimerTask> _timerTasks;
+        private ITimeTrackingStrategy _timeTrackingStrategy;
+        private Dictionary<int, Task> _currentlyRunning;
+        private int _taskCounter = 0;
 
+        // TODO: Decide how to handle config values
+        //
+        private TimeSpan _epsilon = TimeSpan.FromMilliseconds(100);
+        #endregion
+        public TimeSpan ElapsedTime => throw new NotImplementedException();
         public TimeSpan TimeLeft => throw new NotImplementedException();
+        public Guid Id { get; private set; }
 
-        Guid ITimer.Id => throw new NotImplementedException();
+        #region Explicitly implemented internal
+        void ITimer.Initialize(ITimeTrackingStrategy timeTrackingStrategy, List<ITimerTask> timerTasks)
+        {
+            Id = Guid.NewGuid();
+            foreach(ITimerTask timerTask in timerTasks)
+            {
+                // TODO: consider enforcing minimum period.
+                //
+                _timerTasks.Add(timerTask.Period, timerTask);
+            }
 
-        public void Clear()
+            _timeTrackingStrategy = timeTrackingStrategy;
+        }
+        public BasicTimer()
         {
-            throw new NotImplementedException();
+            _timerTasks = new SortedList<TimeSpan, ITimerTask>();
+            _currentlyRunning = new Dictionary<int, Task>();
+            _timeTrackingStrategy = null;
         }
 
-        public bool IsStopped()
+        /// <summary>
+        /// Schedules tasks based on ideal timings. For example, if a task has a period of
+        /// 1s but doesn't get run until T = 1.1s, the next schedule will be for T = 2s 
+        /// rather than 2.1s. Likewise for T = 0.9: second execution would be for T = 2s 
+        /// rather than 1.9.
+        /// 
+        /// Maintains a list of currently running tasks to potentially be cancelled later.
+        /// </summary>
+        /// <returns>Awaitable list of tasks running during this update.</returns>
+        async Task ITimer.Update()
         {
-            throw new NotImplementedException();
-        }
+            if(_timerTasks.Count == 0)
+            {
+                // TODO: Should we throw? Probably shouldn't have dangling Timers
+                //
+                await Task.CompletedTask;
```

Comment:
Return Task.CompletedTask instead of awaiting - that would create a new thread and then return